### PR TITLE
Creating a detailed stats table for capturing details on the execution of dq rules.

### PR DESCRIPTION
**Pull request by vigneshwarrvenkat**
detailed stats table for capturing details on the execution of dq rules

## Description

https://github.com/Nike-Inc/spark-expectations/issues/55

## Related Issue

https://github.com/Nike-Inc/spark-expectations/issues/55

## Motivation and Context

As of now, the stats table details are not sufficient to get detailed insights on the rules executed. This PR provides more details using a detailed stats table on the execution of all the rules in a relational format.

## How Has This Been Tested?

The testing was done in local env with the sample data provided with this framework.

## Screenshots (if appropriate):
image
image
image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: spark_expectations/utils/actions.py [hunk 1]
```diff
@@ -94,6 +97,338 @@ def create_rules_map(_rule_map: Dict[str, str]) -> Any:
...
+
+                        # assignment of actual_outcome and expected_outcome
+                        actual_outcome = _querydq_source_query_output
+                        expected_outcome = (
+                            str(match.group(2)) + str(_querydq_target_query_output)
+                            if _querydq_target_query_output
+                            else None
+                        )
+
+                        # if match.group(1):
```
Comment (asingamaneni) [inline spark_expectations/utils/actions.py]: Can we clean up these commented code please

File: docs/getting-started/setup.md [hunk 2]
```diff
@@ -93,3 +95,35 @@ create table if not exists `catalog`.`schema`.`dq_stats` (
 );
 ```
 
+### DQ Detailed Stats Table
```
Comment (asingamaneni) [inline docs/getting-started/setup.md:98]: Will the data for this stats table also be sent to kafka stream if streaming options are enabled?

Comment (vigneshwarrvenkat) [reply docs/getting-started/setup.md:98]: As of right now, it does not but nothing harm in writing if needed, as it contains only metadata.

File: spark_expectations/utils/actions.py (hunk 1 above)
Comment (vigneshwarrvenkat) [reply spark_expectations/utils/actions.py]: Done

File: docs/getting-started/setup.md (hunk 2 above)
Comment (asingamaneni) [reply docs/getting-started/setup.md:98]: Yeah, We can take it up as a follow-up issue.

**Review by asingamaneni — APPROVED** [review]
Thanks for your contribution.

**Pull request merged**